guides/documentations/essentials.md

```diff
@@ -0,0 +1,214 @@
+
+## Metadata management
+
+A component represents a piece of code, related to a specific service, that is defined via the Metadata. It is the building block for creating a diagram. Some components are connected to each other either by links, or because one component may contain another. These actions can be performed through the 'draw' interface or by modifying component attributes in the component panel (in Leto-modelizer).
+
+The global purpose of Metadata is to list all components of the application, define their content, how they are displayed and specify the rules for their various properties.
+
+In practice, Metadata is divided into two parts:
+
+- Declarative Part: get the JSON file that describes the component in a functional way. The format of the JSON file is defined by the plugin, not by the plugin-core.
+- Technical Part: Metadata classes are used to convert previous JSON into a list of classes, used in the let-modelizer to retrieve the list of components and their rules.
+
+| Left: Metadata - Right: Classes from converted metadata |
+| :--: |
+| <img height="400" src="../../documentations/metadata_storage.png"/> |
+
+Keep in mind that when writing metadata, you are describing the language and how it is supposed to behave visually. For instance, if there is a link between two components, you should include it in the metadata.
+
+Here is the documentation for the [ComponentDefinition](https://ditrit.io/leto-modelizer-plugin-core/ComponentDefinition.html) class.
+As you saw in the previous image, component definitions are loaded from a metadata file. Each component definition has:
+- A name
+- An icon
+- Attributes
+- ...
+
+| Left: Components list in Leto-modelizer - Middle: Related Metadata - Right: Component Panel |
+| :--: |
+| <img height="400" src="../../documentations/metadata.png"/> |
+
+Attributes of a component represent all the related properties that can/must be set inside the Component Panel (Right side of the previous image).
+Link Definitions can also be loaded from the metadata file (for colored links, dashed links...).
+
+Once the component definitions are loaded, components can be instantiated from the components list (Left side of the previous image) in Leto-modelizer.
+The DefaultMetadata class provides two methods:
+
+- `validate`: Validate the metadata.
+- `parse`: Set all component/link definitions in PluginData from your provided metadata.
+
+See example on how metadata from a JSON is used inside of the terraformMetadata class for Terrator-plugin here (https://github.com/ditrit/terrator-plugin/blob/main/src/metadata/TerraformMetadata.js)
+
+## Renderer
+
+The Renderer is used to serialize all the instantiated components into source files (`.tf` files for Terraform, `.yaml` files for Kubernetes, ...). It uses a templating engine to work, the choice depends on the plugin. For example, terrator-plugin and githubator-plugin use [Nunjucks templates](https://mozilla.github.io/nunjucks/).
```

Comment:
`terrator-plugin and githubator-plugin use` -> `terrator-plugin and githubator-plugin are using`

Also maybe link to the repo for both plugin ?

Reply:
done, and added a line with link to both plugin at the top of the file